app/src/main/java/com/stevesoltys/seedvault/restore/RestoreViewModel.kt

```diff
@@ -200,19 +204,13 @@ internal class RestoreViewModel(
             return
         }
 
-        // we need to retrieve the restore sets before starting the restore
-        // otherwise restoreAll() won't work as they need the restore sets cached internally
-        val observer = RestoreObserver { observer ->
-            // this lambda gets executed after we got the restore sets
-            // now we can start the restore of all available packages
-            val restoreAllResult = session.restoreAll(token, observer, monitor)
-            if (restoreAllResult != 0) {
-                Log.e(TAG, "restoreAll() returned non-zero value")
-                mRestoreBackupResult.postValue(
-                    RestoreBackupResult(app.getString(R.string.restore_finished_error))
-                )
-            }
-        }
+        // We need to retrieve the restore sets before starting the restore.
+        // Otherwise, restorePackages() won't work as they need the restore sets cached internally.
+        val packages = mChosenRestorableBackup.value?.packageMetadataMap?.keys?.toList()
+            ?: emptyList()
```

Comment:
Does it make sense to continue with an empty list here? Should we fail with an error instead? Can it happen that we don't get the list of packages? If so, when?

Reply:
Fails with an error now.